portfolio_sim/utils.py

```diff
@@ -1,7 +1,15 @@
 import pandas as pd
 import requests
-import urllib
+import urllib.parse
+import yfinance as yf
 from abc import ABC, abstractmethod
+from requests_cache import CacheMixin, SQLiteCache
+from requests_ratelimiter import LimiterMixin, MemoryQueueBucket
+import numpy as np
+
+
+class CachedLimiterSession(CacheMixin, LimiterMixin, requests.Session):
+    """A class for caching and limiting requests"""
 
 
 class APIRetriever(ABC):
```

Comment:
should the APIReciever logic along with the implementations all be in one file called `utils.py`? Maybe it make sense to extract this logic to another file with a better name